content/en/docs/spin-operator/tutorials/pub-sub-with-redis.md

```diff
@@ -0,0 +1,213 @@
+kubectl apply -f https://github.com/spinkube/spin-operator/releases/download/v0.1.0/spin-operator.runtime-class.yaml
+kubectl apply -f https://github.com/spinkube/spin-operator/releases/download/v0.1.0/spin-operator.shim-executor.yaml
+```
+
+## Install Spin Operator 
+
+Then, we install Spin Operator which handles the `SpinApp` application that we are about to create:
+
+```bash
+helm install spin-operator \
+  --namespace spin-operator \
+  --create-namespace \
+  --version 0.1.0 \
+  --wait \
+  oci://ghcr.io/spinkube/charts/spin-operator
+```
+
+## Redis
+
+Let's dive in and get Redis sorted because we are going to need information about our Redis installation in our Spin application's config. We will use the following `helm` commands to get the job done:
+
+```bash
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm repo update
+helm install my-redis bitnami/redis
+```
+
+The `helm` installation process from above prints a lot of useful information to the terminal. For example, the endpoints to communicate with Redis (read/write vs read-only):
+
+```bash
+my-redis-master.default.svc.cluster.local for read/write operations (port 6379)
+my-redis-replicas.default.svc.cluster.local for read-only operations (port 6379)
+```
+
+In addition, there are pre-written commands that you can cut and paste. For example:
+
+```bash
+export REDIS_PASSWORD=$(kubectl get secret --namespace default my-redis -o jsonpath="{.data.redis-password}" | base64 -d)
+```
+
+Go ahead and run the command above to set the password. And, if need be, you can run the following command to see the actual password printed in your terminal:
+
+```bash
+echo $REDIS_PASSWORD
+```
+
+## Create a Redis Message Handler Using Rust
+
+We use Spin's convenient `redis-rust` template to scaffold our Rust-based Redis message handler:
+
+```bash
+spin new -t redis-rust redis-message-handler
+```
+
+The command above will provide the prompts for you to add the Description, Redis address and Redis channel (We use the `my-redis-master.default.svc.cluster.local` from above to help configure the Redis address, and the channel is arbitrary i.e. `channel-one`):
+
+```bash
+Description: Redis message handler using Rust
+Redis address[redis://localhost:6379]: redis://my-redis-master.default.svc.cluster.local:6379
+Redis channel: channel-one
+```
+
+## Configure Spin Application
+
+We change into our application directory, and can see the layout that Spin has scaffolded for us:
+
+```bash
+cd redis-message-handler
+tree .
+```
+
+The above `tree .` command, produces the following output:
+
+```bash
+.
+├── Cargo.toml
+├── spin.toml
+└── src
+    └── lib.rs
+```
+
+If we open the application manifest (`spin.toml` file) we see that Spin has already pre-populated the [Redis trigger configuration](https://developer.fermyon.com/spin/v2/redis-trigger#the-spin-redis-trigger):
+
+```toml
+// --snip --
+[application.trigger.redis]
+address = "redis://my-redis-master.default.svc.cluster.local:6379"
```

Comment:
As of Spin v2.4.0, the redis address can access application variables for its value, so this could change to 

```suggestion
address = "{{ redis_address}}"
```
assuming the following section was also included in the manifest:
```toml
[variables]
redis_address = { required = true }
```
However, since this is running on K8s using the [containerd-shim-spin](https://github.com/spinkube/containerd-shim-spin), we'd not want to show this feature until the next version of the shim is out.  (v0.13.1 uses Spin v2.3.1 which lacks this feature but the main branch does already have v2.4.2.)

See also the following PR in the shim project updating its redis trigger example: https://github.com/spinkube/containerd-shim-spin/pull/78